pkg/api/steve/proxy/proxy.go

```diff
@@ -213,14 +213,14 @@ func (h *Handler) dialer(ctx context.Context, network, address string) (net.Conn
 }
 
 func (h *Handler) next(clusterID, prefix string) (http.Handler, error) {
+	ht := http.DefaultTransport.(*http.Transport).Clone()
```

Comment:
Possibly a stupid question but the default transport [will make use of HTTPS_PROXY / HTTP_PROXY](https://pkg.go.dev/net/http#DefaultTransport) vars if they are set. We set those in the [Rancher deployment](https://github.com/rancher/rancher/blob/4a27e55ae0a51ca6b75d0c6aaadcb2362688acb0/chart/templates/deployment.yaml#L125-L132). 

Would this prevent the connection from being established?

Reply:
Definitely not stupid! I don't know how the `Proxy` settings come into play when providing your own `Dialer` (which in this case will not make use of the network but `remotedialer`).
It makes sense to play safe and remove the default proxy configuration.
Thanks!
```suggestion
	ht := http.DefaultTransport.(*http.Transport).Clone()
	ht.Proxy = nil
```